test/tests/options.rb

```diff
@@ -0,0 +1,291 @@
+    :interactive_ignore     => "--interactive-ignore",
+    :report_routes          => "--routes",
+    :show_version           => "--version",
+    :show_help              => "--help"
+  }
+
+  def test_easy_options
+    EASY_OPTION_INPUTS.each_pair do |key, value|
+      options = setup_options_from_input(value)
+      assert options[key], "Expected #{key} to be #{!!value}."
+    end
+  end
+
+  def test_alt_easy_options
+    ALT_OPTION_INPUTS.each_pair do |key, value|
+      options = setup_options_from_input(value)
+      assert options[key], "Expected #{key} to be #{!!value}."
+    end
+  end
+
+  def test_assume_routes_option
+    options = setup_options_from_input("-a")
+    assert options[:assume_all_routes]
+
+    options = setup_options_from_input("--assume-routes")
+    assert options[:assume_all_routes]
+
+    options = setup_options_from_input("--no-assume-routes")
+    assert options[:assume_all_routes] == false
+  end
+
+  def test_faster_options
+    options = setup_options_from_input("--faster")
+    assert options[:ignore_ifs] && options[:skip_libs]
+  end
+
+  def test_index_libs_option
+    options = setup_options_from_input("--index-libs")
+    assert options[:index_libs]
+
+    options = setup_options_from_input("--no-index-libs")
+    assert options[:index_libs] == false
+  end
+
+  def test_limit_options
+    options = setup_options_from_input("--branch-limit", "17")
+    assert options[:branch_limit] == 17
+  end
+
+  def test_no_threads_option
+    options = setup_options_from_input("-n").merge!({
+      :quiet => true,
+      :app_path => "#{TEST_PATH}/apps/rails4"})
+
+    assert !options[:parallel_checks], "Expected app to be running threaded"
+  end
+
+  def test_path_option
+    options = setup_options_from_input("--path", "#{TEST_PATH}/app/rails4")
+    assert options[:app_path], "#{TEST_PATH}/app/rails4"
+
+    options = setup_options_from_input("-p", "#{TEST_PATH}/app/rails4")
+    assert options[:app_path], "#{TEST_PATH}/app/rails4"
+  end
+
+  def test_progress_option
+    options = setup_options_from_input("--progress")
+    assert options[:report_progress]
+
+    options = setup_options_from_input("--no-progress")
+    assert options[:report_progress] == false
+  end
+
+  def test_quiet_option
+    options = setup_options_from_input("-q")
+    assert options[:quiet]
+
+    options = setup_options_from_input("--quiet")
+    assert options[:quiet]
+
+    options = setup_options_from_input("--no-quiet")
+    assert options[:quiet] == false
+  end
+
+  def test_rails_4_option
+    options = setup_options_from_input("-4")
+    assert options[:rails4] && options[:rails3]
+
+    options = setup_options_from_input("--rails4")
+    assert options[:rails4] && options[:rails3]
+  end
+
+  def test_safe_methods_option
+    options = setup_options_from_input("--safe-methods", "test_method2,test_method1,test_method2")
+    assert options[:safe_methods] == Set.new([:test_method1, :test_method2])
```

Comment:
When doing comparisons, `assert_equal` would be the typical approach (and produces better failure messages). Expected value comes first.

And you can use `Set[]` to create the set.

```ruby
assert_equal Set[:test_method1, :test_method2], options[:safe_methods]
```